### DISCUSSION TITLE
How to Approach Bush Rustles on Interaction

### DISCUSSION THREAD

**Discussion by Lionorben** [visualdesigncafe/nature-renderer#50] [Help]
Ive got a world that is coming along very well, however one immersion breaking thing atm is that there are no sounds that get triggered when walking through bushes, footstep sounds environment sounds and all that are in, but this is a piece that is still missing. Im wondering how i should approach this. Can i tie into the interaction system to call a function when interacting with objects? should i be baking trigger colliders and using a custom script? Performance wise linking to the interaction system seems ideal, any help is greatly appreciated ^-^

~Ben

-heres a snip of the game so far ;) 
![snip](https://github.com/visualdesigncafe/nature-renderer/assets/64295361/939fdab8-a9b3-4075-a717-d3dc6db706a5)

**Comment by VeniaminKarach** [comment]
looks awesome, do u have a page on Steam with your game already? And what did u use for placing tree&grass instances through the map? Gaia proceduaral or smth else?

**Reply by Lionorben** [reply]
Holy moly someone actually saw this! I work at PW so i used Gaia for placement, steam page is currently under a major overhaul with these new graphics so it looks pretty bad but the game is called Spriggan! We made a devlog last year before the graphics reboot: https://youtu.be/HkRjbuy54PU

~Ben

**Comment by maxplooi** [comment]
The interaction system is fully handled on the GPU in shaders, so there is no way to access that data via script. I recommend to sample the vegetation placement from the terrain via the regular terrain API to compute what vegetation the player is standing on and then use that to play audio.

**Reply by acorsicanfrog** [reply]
Hello Ben,

As Max said, this isn't something Nature Renderer is supposed to cover.
However, I will help you getting this working as I have done it for my game already.

Here is my script which handles spawning colliders for all vegetation (trees and bushes): https://pastebin.com/s6Pww3eW

You can adapt it to your needs, but that's basically what you are gonna need

It's not difficult :)
